### PR TITLE
Jackson's Changes for Notifications

### PR DESCRIPTION
**Pull request by jaxoncreed**
#### Description
Jackson's Changes to Individdata's notifications
 - Some configuration modifications
 - Unsubscribe route
 - Test updates

### CODE REVIEW

**Comment by ixuz** [comment]
Awesome! Thank you @jaxoncreed!
I'll have a look at this tomorrow with our Individdata team.

**Review by TamSzaGot — COMMENTED** [review]
It nice that you have merged your version of building well known handling with WellKnownBuilder.

Missing unit tests for WebHookSubscription2021UnsubscribeHttpHandler.

Your comment in src/http/WebHookSubscription2021UnsubscribeHttpHandler.ts  line 58 about using the wrong data structure for the notificationData.subscriptions may be valid, but our initial take to use a HashMap were not serializable and could not be put into the storage. Please make a better suggestion.

**Review by TamSzaGot — COMMENTED** [review]
Perhaps it would be good to have a link to the notification proposal

https://github.com/solid/notifications-panel/blob/main/proposals/solid-webhook-notifications.md

Into a comment in the classes implementing WebHookSubscription2021.

File: src/notification/webhook-subscription-2021/webhook-auth/WebHookAuthHttpClient.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  public call(
+    url: string | URL,
+    options: RequestOptions,
+    data: any,
+    callback?: ((res: IncomingMessage) => void) | undefined,
+  ): void {
+    const parsedUrl = url instanceof URL ? url : new URL(url);
+    this.jwksKeyGenerator
```
Comment (ixuz) [inline src/notification/webhook-subscription-2021/webhook-auth/WebHookAuthHttpClient.ts:36]: Perhaps this whole promise-chain could be better implemented using async/await.

**Comment by ixuz** [comment]
Hi @jaxoncreed I think this PR might have been merged/closed prematurely. Did you have unfinished work for this PR?
In that case, shall I revert the merge?

**Comment by jaxoncreed** [comment]
No, @ixuz. I don't have unfinished work here. Though, it should be noted that the changes in this branch don't have full test coverage.

**Comment by ixuz** [comment]
Understood!
I previously tried to merge/pull changes from `upstream` it resulted in an issue with importing the `jose` dependencies.
It looks like there's some code in this branch that was written for an older version of `jose` than `4.0.0`.
The breaking changes in `jose` are mentioned here: https://github.com/panva/jose/blob/main/CHANGELOG.md#400-2021-10-14

Are there any particular reason to keep using `jose` `3.x.x` version? If not, we can adopt the latest version.

**Comment by jaxoncreed** [comment]
I don't think so. I don't think I actively made the decision to use 3.x.x

**Pull request merged**